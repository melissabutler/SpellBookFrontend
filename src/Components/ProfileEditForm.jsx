import React, { useContext } from "react";

import useFields from "../Hooks/useFields";
import CurrentUserContext from "../currentUserContext";
import { useNavigate } from "react-router-dom";

import "./Form.css"

import { Col, Row, Container } from 'react-bootstrap';

const ProfileEditForm = ({editUser}) => {
    const navigate = useNavigate();
    let currentUser = useContext(CurrentUserContext);

    const [formData, handleChange, resetFormData] = useFields({
        "email": currentUser.email
    })

    const handleSubmit = e => {
        e.preventDefault();

        let updatedUser = {
            "email": formData.email.toString(),
        }
        editUser({updatedUser})
        navigate('/profile')
    }

    return(
        <Container className="Form">
            <Row>
                <h2>Update User Email</h2>
            </Row>
            
            <form className="Form-form" onSubmit={handleSubmit}>
                <Row>
                    <Col></Col>
                    <Col>
                   <label htmlFor="email" className="Form-label">Email</label> 
                   </Col>
                   <Col></Col>
                </Row>
                <Row>
                    <input
                    className="Form-input"
                    id="email"
                    name="email"
                    type="text"
                    value={formData.email}
                    onChange={handleChange}
                    placeholder="Email Address"
                    /> 
                   
                </Row>
                
         
                <Row>
                    <Col></Col>
                    <Col>
                    <button>Submit</button>
                    </Col>
                    <Col></Col>
                    
                </Row>
                
                
            </form>
        </Container>
    )
     }

export default ProfileEditForm;