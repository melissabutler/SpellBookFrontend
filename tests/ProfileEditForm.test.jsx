import React from "react";
import { describe, it, expect, vi } from "vitest";
import { render, screen} from '@testing-library/react'
import ProfileEditForm from "../src/Components/ProfileEditForm";

describe("ProfileEditForm", function() {

        vi.mock('react', async (importOriginal) => {
            const react = await importOriginal('react');
            return {
                ...react,
                useContext: () => ({ currentUser: null}),
                useRef: () => { target: null }
        }})

        vi.mock('react-router-dom', async () => {
            const mod = await vi.importActual('react-router-dom');
            return {
                ...mod, 
                useNavigate: () => "/",
                BrowserRouter: () => "",
            }
        })
    it('renders theProfileEditForm component', async () => {
        render( <ProfileEditForm />)
        screen.debug();
      
    });

        it('matches snapshot', () => {
            const { asFragment } = render(<ProfileEditForm />);
            expect(asFragment()).toMatchSnapshot();
        });



})
   