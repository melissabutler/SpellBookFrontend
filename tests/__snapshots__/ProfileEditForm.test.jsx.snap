// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ProfileEditForm > matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="Form container"
  >
    <div
      class="row"
    >
      <h2>
        Update User Email
      </h2>
    </div>
    <form
      class="Form-form"
    >
      <div
        class="row"
      >
        <div
          class="col"
        />
        <div
          class="col"
        >
          <label
            class="Form-label"
            for="email"
          >
            Email
          </label>
        </div>
        <div
          class="col"
        />
      </div>
      <div
        class="row"
      >
        <input
          class="Form-input"
          id="email"
          name="email"
          placeholder="Email Address"
          type="text"
          value=""
        />
      </div>
      <div
        class="row"
      >
        <div
          class="col"
        />
        <div
          class="col"
        >
          <button>
            Submit
          </button>
        </div>
        <div
          class="col"
        />
      </div>
    </form>
  </div>
</DocumentFragment>
`;
